### PR TITLE
farwicls_codeReview1

### PR DESCRIPTION
**Pull request by loganfarwick**
Analysis of the program:
Logan Farwick reviewing NewsNow – https://github.com/tron561/IT3048C-Repo/tree/Application 
NewsNow is a mobile application intended to give users an article a day to read, share with others, and to display users their daily streak of reading articles. 

Was the program available in UC GitHub on time?
Yes

Is the program documented/commented well enough for you to understand?
There are very few comments outside of the MainActivity.kt where there are enough for me to understand what given code is doing and its purpose. 

Does the program compile?
Yes

Rationale behind your changes:
1.	Through changing the format of string concatenation, code is more readable and takes out unnecessary characters/spaces/etc. 
2.	Variable used only in following return can be inlined, so as to remove an unnecessary line of code and put it all in one return statement. 
3.	Removed unnecessary qualifier name for retrofit2.Retrofit.Builder() -> Retrofit.Builder(). Unnecessary verbiage/characters add bulk to code. 

Sources:
1.	https://kotlinlang.org/docs/java-to-kotlin-idioms-strings.html
2.	https://kotlinlang.org/docs/inline-functions.html
3.	https://www.baeldung.com/retrofit-dynamic-url 

Links to three specific commits that you made to your own group's GitHub repository before the end of sprint deadline.
1.	https://github.com/jakeFarrish/MedMapper/commit/a819e5d4cdc3568eb8bc0a4741084862fd5f9293
2.	https://github.com/jakeFarrish/MedMapper/commit/c52c6160ea52ff99f66a5bb3bbcd4c9d3f276dfd
3.	https://github.com/jakeFarrish/MedMapper/commit/64c0d1cc17c738ea043b0749761a2d5bf03ac1bf

Three specific technical concepts that you learned from this code review:
•	I got reviewing experience looking over another person/group’s code. I appreciate the ability to see how others go about implementing:
o	Retrofit and utilizing it for different APIs.
o	MainViewModel and how it is utilized in the MainActivity.
o	Koin and the implementation of it in a different setting than PlantPlaces.

### CODE REVIEW

File: app/src/main/java/com/newsnow/service/ArticleService.kt [hunk 1]
```diff
@@ -17,9 +17,8 @@ class ArticleService : IArticleService {
     override suspend fun fetchArticles() : List<Article>? {
         return withContext(Dispatchers.IO) {
             val service = RetrofitClientInstance.retrofitInstance?.create(IArticleDAO::class.java)
-            val countries = async {service?.getAllArticles()}
-            var result = countries.await()?.awaitResponse()?.body()
-            return@withContext result
+            val countries = async { service?.getAllArticles() }
+            return@withContext countries.await()?.awaitResponse<ArrayList<Article>>()?.body()
```
Comment (loganfarwick) [inline app/src/main/java/com/newsnow/service/ArticleService.kt:21]: Variable used only in following return can be inlined, so as to remove an unnecessary line of code and put it all in one return statement. 

https://kotlinlang.org/docs/inline-functions.html

File: app/src/main/java/com/newsnow/RetrofitClientInstance.kt [hunk 2]
```diff
@@ -12,7 +12,7 @@ object RetrofitClientInstance {
             // has this object been created yet?
             if (retrofit == null) {
                 // create it
-                retrofit = retrofit2.Retrofit.Builder()
+                retrofit = Retrofit.Builder()
```
Comment (loganfarwick) [inline app/src/main/java/com/newsnow/RetrofitClientInstance.kt:15]: Removed unnecessary qualifier name for retrofit2.Retrofit.Builder() -> Retrofit.Builder(). Unnecessary verbiage/characters add bulk to code. 

https://www.baeldung.com/retrofit-dynamic-url

File: app/src/main/java/com/newsnow/dto/Article.kt [hunk 3]
```diff
@@ -15,6 +14,6 @@ data class Article(
 ) {
     // returns article title and full description
     override fun toString(): String {
-        return title + " " + full_description
+        return "$title $full_description"
```
Comment (loganfarwick) [inline app/src/main/java/com/newsnow/dto/Article.kt:17]: Through changing the format of string concatenation, code is more readable and takes out unnecessary characters/spaces/etc. 

https://kotlinlang.org/docs/java-to-kotlin-idioms-strings.html

Comment (discospiff) [reply app/src/main/java/com/newsnow/dto/Article.kt:17]: While you're in here... full_description does not use proper naming conventions.  Fix that.

File: app/src/main/java/com/newsnow/service/ArticleService.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/com/newsnow/service/ArticleService.kt:21]: Well... you could also say it's more readable on three lines.

Formatting can, and should, be performed automatically, and consistently, by the IDE.  Formatting code is really programmer's preference anyway, and doesn't necessarily decrease technical debt.

**Comment by discospiff** [comment]
I'm neutral on merging this, will leave it up to the team.  The changes here are fairly minor, and don't have a significant impact on technical debt.

**Pull request closed without merging**